s3/src/main/scala/akka/stream/alpakka/s3/S3Settings.scala

```diff
@@ -26,15 +31,19 @@ object S3Settings {
   /**
    * Create [[S3Settings]] from a Config subsection.
    */
-  def apply(config: Config): S3Settings =
+  def apply(config: Config): S3Settings = {
+    val proxyConfig = if (config.hasPath("proxy")) config.getConfig("proxy") else ConfigFactory.empty()
```

Comment:
The problem with not defining "all" properties in `reference.conf` is that it's difficult for the user to know about their existence. Yeah, it could be documented, but I think it's nice to have them in reference.conf, even if it means having a special value for undefined.

could we use something like:
```
proxy {
  # Hostname of the proxy. If undefined ("") proxy is not enabled.
  host = ""
  port = 8000
}